pkgs/development/tools/build-managers/bazel/bazel_7/nix-build-bazel-package-hacks.patch

```diff
@@ -1,46 +1,37 @@
 diff --git a/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java b/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java
-index 4d8c46f8d5..ed311226f0 100644
+index 53e6494656..22261cd268 100644
 --- a/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java
 +++ b/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java
-@@ -178,18 +178,8 @@ public final class RepositoryDelegatorFunction implements SkyFunction {
-         }
+@@ -55,6 +55,7 @@ import com.google.devtools.build.skyframe.SkyFunctionException.Transience;
+ import com.google.devtools.build.skyframe.SkyKey;
+ import com.google.devtools.build.skyframe.SkyValue;
+ import java.io.IOException;
++import java.util.Collections;
+ import java.util.Map;
+ import java.util.Optional;
+ import java.util.TreeMap;
+@@ -193,16 +194,11 @@ public final class RepositoryDelegatorFunction implements SkyFunction {
        }
+ 
        if (shouldUseCachedRepos(env, handler, repoRoot, rule)) {
 -        // Make sure marker file is up-to-date; correctly describes the current repository state
 -        byte[] markerHash = digestWriter.areRepositoryAndMarkerFileConsistent(handler, env);
 -        if (env.valuesMissing()) {
 -          return null;
 -        }
 -        if (markerHash != null) { // repo exist & up-to-date
--          return RepositoryDirectoryValue.builder()
--              .setPath(repoRoot)
++        {
++          // Nix hack: Always consider cached dirs as up-to-date
+           return RepositoryDirectoryValue.builder()
+               .setPath(repoRoot)
 -              .setDigest(markerHash)
--              .setExcludeFromVendoring(shouldExcludeRepoFromVendoring(handler, rule))
--              .build();
--        }
-+        // Nix hack: Always consider cached dirs as up-to-date
-+        return RepositoryDirectoryValue.builder().setPath(repoRoot).setDigest(digestWriter.writeMarkerFile()).build();
-       }
- 
-       /* At this point: This is a force fetch, a local repository, OR The repository cache is old or
-@@ -610,11 +600,12 @@ public final class RepositoryDelegatorFunction implements SkyFunction {
-         builder.append(escape(key)).append(" ").append(escape(value)).append("\n");
+-              .setExcludeFromVendoring(excludeRepoFromVendoring)
++              .setDigest(digestWriter.writeMarkerFile(Collections.emptyMap()))
+               .build();
+         }
```

Comment:
Have you tested if the patch works? It seems to generate an extra scope:

```java
if (shouldUseCachedRepos(env, handler, repoRoot, rule)) {
  {
    // Nix hack: Always consider cached dirs as up-to-date
    return RepositoryDirectoryValue.builder()
        .setPath(repoRoot)
        .setDigest(digestWriter.writeMarkerFile(Collections.emptyMap()))
        .build();
  }
}
```

Does this actually return from the function or just from the scope (not really familiar with Java).

Reply:
Extra nested block creates a nested lexical scope but `return` still refers to the enclosing function as there's no block-level returns. Added it to reduce formatting indentation diff against bazel upstream